server/database/handle-goal-data.js

```diff
@@ -1,41 +1,63 @@
 const dbClient = require('./db_connection.js');
 
-const getGoal = `
-  SELECT *
-    from goals
-  WHERE goal_id = $1
-`;
-
-const addGoal = `
-  INSERT into goals (goal_id, user_id, title, icon, date_created)
-  VALUES ($1, $2, $3, $4, $5)
-  RETURNING goal_id
-`;
+/* queries */
+const queries = require('./queries.js');
+const addRatings = require('./add-ratings.js');
+
+/* helper functions */
+const { getNewGoalData, findNewRatings, } = require('./../helpers/handle-goals.js');
+
+// compares new goal from client with old goal from DB, adds missing ratings to database
+const updateGoal = (dbGoal, clientGoal, callback) => {
+
+  // check for new ratings
+  dbClient.query(queries.getRatings, [ clientGoal.id, ], (dbRatingsErr, dbRatingsRes) => {
```

Comment:
Generally it is hard to work out what is going on here. What I do
queries (plain text) -> queryFunctions -> controllers

ie
queries.js
```js
const q1 = `...`
const q2 = `...`
```
model.js
```js
import queries, pg.query
const getThis = (data, cb)
const getThat = (data, cb)
const moreComplicated = (data, cb) // this might compose `getThis` and `getThat` internally
```
controller.js
```js
  reply(async.waterfall(getThis, moreComplicated)
```
**Business logic lives in the controller(handler)** -> you want to build up the functionality under this in layers